### PR TITLE
feat: support separate config of `hoverHighlight`

### PR DESCRIPTION
**Pull request by tyr1dev**
### 👀 PR includes

✨ Feature

- [x] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [x] Test case
- [x] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [x] Other (s2 react playground) 

### 📝 Description

更细致的 `hoverHighlight` 配置

### 🔗 Related issue link

close: #2225 

### 🔍 Self-Check before the merge

- [x] Add or update relevant docs.
- [x] Add or update relevant demos.
- [x] Add or update test case.
- [x] Add or update relevant TypeScript definitions.

👇🏻👇🏻👇🏻
⚠️ 可能会造成 break change，如果用户使用 checked 开关并绑定 `hoverHighlight` 可能会导致 TS error

### CODE REVIEW

**Comment by tyr1dev** [comment]
A bit strange, test passed on my machine.

File: s2-site/docs/common/interaction.zh.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ order: 5
...
 | linkFields  | 标记字段为链接样式，用于外链跳转  | `string[]` \| (meta: [Node](/docs/api/basic-class/node) \| ViewMeta) => boolean    |    |    |
 | selectedCellsSpotlight               | 是否开启选中高亮聚光灯效果    | `boolean`                                 | `false`      |                  |
-| hoverHighlight                       | 鼠标悬停时高亮当前单元格，以及所对应的行头，列头                                                                                                               | `boolean`                                 | `true`                                                |                  |
+| hoverHighlight                       | 鼠标悬停时高亮当前单元格，以及所对应的行头，列头<br/>rowHeader：是否高亮悬停格子所在行头<br/>colHeader：是否高亮悬停格子所在列头<br/>rowCells：是否高亮悬停格子所在行<br/>colCells：是否高亮悬停格子所在列<br/>true：同 `{rowHeader: true, colHeader: true, currentRow: true, currentCol: true}`                                                                                                                 | `boolean \| { rowHeader?: boolean, colHeader?: boolean, rowCells?: boolean, colCells?: boolean}`                                 | `true`                                                |                  |
```
Comment (lijinke666) [inline s2-site/docs/common/interaction.zh.md:12]: rowCells => currentRow ?
colCells => currentCol ?

Comment (tyr1dev) [reply s2-site/docs/common/interaction.zh.md:12]: 代码里一直是 `currentXXX`，文档没更新

Comment (lijinke666) [reply s2-site/docs/common/interaction.zh.md:12]: 这个 demo s2-site/examples/interaction/basic/demo/hover.ts 建议增加下配置的示例 (注释说明一下就好)

交互章节 `s2-site/docs/manual/advanced/interaction/basic.zh.md ### 行列联动高亮` 也可以说明一下

image

File: packages/s2-core/src/common/constant/interaction.ts [hunk 2]
```diff
@@ -124,3 +124,10 @@ export interface InteractionCellSelectedHighlightType {
   currentRow?: boolean; // 高亮选中单元格所在行
   currentCol?: boolean; // 高亮选中单元格所在列
 }
+
+export interface HoverHighlight {
```
Comment (lijinke666) [inline packages/s2-core/src/common/constant/interaction.ts:128]: image

这个两个摞到 packages/s2-core/src/common/interface/interaction.ts 中吧, `constant/xx` 下放枚举和常量

`packages/s2-core/src/common/constant/interaction.ts` => `packages/s2-core/src/common/interface/interaction.ts`

另外, 没必要新增类型吧, 两个是一回事, 直接 `InteractionCellSelectedHighlightType` => `InteractionCellHighlight`

**Comment by lcx-seima** [comment]
辛苦 rebase 一下 master 推送一下最新代码~

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**